MHVideoPhotoGallery/MMHVideoPhotoGallery/MHGalleryImageViewerViewController/MHGalleryImageViewerViewController.m

```diff
@@ -499,26 +501,59 @@ - (void)downloadFile:(MHGalleryItem *)item {
-            [weakSelf showShareDialog: imageViewController.imageView.image];
-        } else {
-            [weakSelf downloadFile:imageViewController.item];
+    self.isAttachmentUploaded = NO;
+    MHImageViewController *imageViewController = (MHImageViewController*)self.pageViewController.viewControllers.firstObject;
+    NSString * URLString = imageViewController.item.URLString;
+    NSURL *fileURL = [NSURL URLWithString:URLString];
+    self.isAttachmentUploaded = ![fileURL isFileURL];
+    
+    if (self.isAttachmentUploaded) {
+        UIAlertController *actioSheet = [UIAlertController alertControllerWithTitle:MHGalleryLocalizedString(@"Choose action")
+                                                                            message:nil preferredStyle:UIAlertControllerStyleActionSheet];
+        __weak typeof(self) weakSelf = self;
+        [actioSheet addAction:[UIAlertAction actionWithTitle:MHGalleryLocalizedString(@"Share Link") style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
+            MHImageViewController *imageViewController = (MHImageViewController*)weakSelf.pageViewController.viewControllers.firstObject;
+            [weakSelf showShareDialog: imageViewController.item.URLString];
+            
+            NSString * str = imageViewController.item.URLString;
+            NSLog(@"");
+        }]];
+        [actioSheet addAction:[UIAlertAction actionWithTitle:MHGalleryLocalizedString(@"Export Document") style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
+            MHImageViewController *imageViewController = (MHImageViewController*)weakSelf.pageViewController.viewControllers.firstObject;
+            if (imageViewController.item.galleryType == MHGalleryTypeImage) {
+                [weakSelf showShareDialog: imageViewController.imageView.image];
+            } else {
+                [weakSelf downloadFile:imageViewController.item];
+            }
+        }]];
+        [actioSheet addAction:[UIAlertAction actionWithTitle:MHGalleryLocalizedString(@"shareview.download.cancel") style:UIAlertActionStyleCancel handler:nil]];
+        if (actioSheet.popoverPresentationController) {
+            actioSheet.popoverPresentationController.barButtonItem = self.shareBarButton;
         }
-    }]];
-    [actioSheet addAction:[UIAlertAction actionWithTitle:MHGalleryLocalizedString(@"shareview.download.cancel") style:UIAlertActionStyleCancel handler:nil]];
-    if (actioSheet.popoverPresentationController) {
-        actioSheet.popoverPresentationController.barButtonItem = self.shareBarButton;
+        [self presentViewController:actioSheet animated:YES completion:nil];
+    } else {
+        NSString *title = @"Whoops...";
+        NSString *message = @"The file hasn't been uploaded yet";
+        
+        UIAlertController *alertController = [UIAlertController alertControllerWithTitle:MHGalleryLocalizedString(title)
```

Comment:
@ViachaslauHolikau strings are not localized in MHGalleryLocalizedString MHGallery.strings